ggml/src/ggml-cpu/ggml-cpu.c

```diff
@@ -8515,7 +8515,9 @@ static void ggml_compute_forward_get_rows_f32(
         const int64_t i10 = (i - i12*ne11*ne10 - i11*ne10);
         const int64_t i01 = *(int32_t *) ((char *) src1->data + i10*nb10 + i11*nb11 + i12*nb12);
 
-        GGML_ASSERT(i01 >= 0 && i01 < ne01);
+        // Copying this out for a bit while investigating due to issues like:
+        // https://github.com/ggerganov/llama.cpp/issues/10157
+        // GGML_ASSERT(i01 >= 0 && i01 < ne01);
```

Comment:
Is this still needed?

Reply:
Hi @ggerganov, thank you very much for your thoughts!

For now, on CPU it is - commenting it out does run & give coherent outputs on CPU, but I think that there is a bug here for siglip, which I am guessing is an issue of how visual encoders without the `CLS` embedding are handled, since the models that I have seen raise this issue appear to use siglip as the visual encoder.

I think I am getting close to the actual fix - I'll follow up (with hopefully a correct fix and this assert reenabled) by the end of today!

Reply:
Opened a separate small PR to fix the underlying cause here! https://github.com/ggml-org/llama.cpp/pull/11982

Will reenable this assertion on this PR now.